xwiki-rendering-macros/xwiki-rendering-macro-html/src/main/java/org/xwiki/rendering/internal/macro/html/HTMLMacro.java

```diff
@@ -152,83 +139,29 @@ public List<Block> execute(HTMLMacroParameters parameters, String content, Macro
                 normalizedContent = renderWikiSyntax(normalizedContent, context.getTransformation(), context);
             }
 
-            // Clean the HTML into valid XHTML if the user has asked (it's the default).
-            if (parameters.getClean()) {
-                normalizedContent = cleanHTML(normalizedContent, context);
-            } else if (context.getTransformationContext().isRestricted()) {
-                throw new MacroExecutionException(
-                    "The HTML macro may not be used with clean=\"false\" in this context.");
+            RawBlock contentBlock = new RawBlock(normalizedContent, targetSyntax);
+
+            try {
+                RawBlockFilterParameters filterParameters = new RawBlockFilterParameters();
+                filterParameters.setClean(parameters.getClean());
+                filterParameters.setRestricted(parameters.getRestricted());
+                filterParameters.setMacroTransformationContext(context);
+                for (RawBlockFilter filter
+                    : this.componentManager.<RawBlockFilter>getInstanceList(RawBlockFilter.class)) {
+                    contentBlock = filter.filter(contentBlock, filterParameters);
+                }
+            } catch (ComponentLookupException e) {
+                throw new MacroExecutionException("Couldn't initialize the HTML filtering.", e);
```

Comment:
This block looks duplicated with https://github.com/xwiki/xwiki-rendering/pull/219/files#diff-62bc96d84833c84f3d85d8f3b8f5fe40eb9b8dece414df58660f576f2c6cc8ecR91-R101